File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const map = new Map();
        var s=[];
        for(var i=0;i<nums.length;i++){
            const diff = target - nums[i];
            console.log(diff);
            if(map.has(diff)){
                s.push(i,map.get(diff));
            }
            else{
                map.set(nums[i],i);
            }
        }
        return s;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        #brute - 3 nested for loops
        #optimal -> find 2sum and check for difference of that number

        result = set()
        for i in range(len(nums)):
            seen=set()
            for j in range(i+1,len(nums)):
                target = -(nums[i]+nums[j])
                if target in seen:
                    triplet = tuple(sorted([nums[i],nums[j],target]))
                    result.add(triplet)
                seen.add(nums[j])

        return list(result)
            
            